### ISSUE TITLE
Default behaviour of config file

### ISSUE THREAD

**Issue by talal** [jwilm/alacritty#1872]
This is not an issue but rather a question to better understand the default behaviour of the config file.

Is it necessary to have all the keybindings configured in the config file (as they are stated in the default config files), or can I remove them and only keep the ones that I have changed?

What I mean is that, does Alacritty have default hardcoded keybindings for different platforms like Linux and macOS which work in the same way as other configuration options.
For example the boolean configuration options need to be only specified in the config file, if the user needs to set them to `true`, the default value of `false` is implied and in that case the option can be removed from the config file. I am assuming this is the default behavior?

Another example would be the `selection.semantic_escape_chars`, can it be removed from the config file if the user has not changed it?

**Comment by chrisduerr** [comment]
The keybindings section is pretty much the only setting which is not in its default state in the default configuration file.

This should be resolved with https://github.com/jwilm/alacritty/issues/40, there have been some discussions on how to best resolve this, but it will likely involve adding some default keybindings for Alacritty and a way to disable this default behavior for certain keys.

Just as an example of how little is actually required, [this](https://github.com/chrisduerr/dotfiles/blob/master/files/alacritty.yml) is my Alacritty configuration file. As you can see, there's not much other than keybindings.

Please let me know if you have any more questions, I'm happy to answer them even after the issue has been closed.

**Comment by talal** [comment]
Thanks for the quick response. This clears up the keybinding part, what about settings like `selection.semantic_escape_chars` and `visualbell`, you have those in your config file even though they don't differ from their default values.

Also for my understanding, when you say:
> The keybindings section is pretty much the only setting which is not in its default state in the default configuration file.

Does that mean that Alacritty uses a default configuration file that is unrelated to the paths [listed here](https://github.com/jwilm/alacritty#configuration), rather than defaults existing in the codebase itself which get overridden by the user's config file.

If yes, then where is the default config file located in contrast to the user's config file? 

Sorry for all the questions 😅. I don't anything about Rust, otherwise I would have scanned the codebase myself to see the concerning functionality behind the scenes.

**Comment by chrisduerr** [comment]
If there is no configuration file provided, Alacritty will use the default configuration file which is compiled into the application. This is identical to the `alacritty.yml` (and the platform-specific variations) in the repository, it's just hardcoded into the binary so we don't have to distribute the file.

As I've added in my `Note` at the end of my message, the `selection.semantic_escape_chars` are actually required right now and empty by default, however there's no good reason for that. Opening a separate issue to track and resolve that would probably be a good idea.

I've set the `visual_bell.duration` simply to disable the visual bell, which is enabled by default.
